programs/zstdcli.c

```diff
@@ -369,14 +394,27 @@ static unsigned readU32FromChar(const char** stringPtr) {
  *  Will also modify `*stringPtr`, advancing it to position where it stopped reading.
  *  Note : function will exit() program if digit sequence overflows */
 static int readIntFromChar(const char** stringPtr) {
-    static const char errorMsg[] = "error: numeric value overflows 32-bit int";
+    
     int sign = 1;
     unsigned result;
+    const char* startChar;
+    int returnVal;
+    
     if (**stringPtr=='-') {
         (*stringPtr)++;
         sign = -1;
     }
-    if (readU32FromCharChecked(stringPtr, &result)) { errorOut(errorMsg); }
+
+    startChar = *stringPtr;
+    
+    returnVal = readU32FromCharChecked(stringPtr, &result);
+    
+    if (returnVal == 1) {
+        errorOut("error: numeric value overflows 32-bit int");
+    }
```

Comment:
The overflow check here isn't good enough as signed and unsigned integers have different ranges.
While I understand this is the behavior that was implemented before, it would be good to augment the overflow checks with comparisons to `INT_MAX` and `INT_MIN`. 
An easy way to implement these checks would be to convert to i64 and compare the 64 bit value.

Reply:
Hi @yoniko - agreed, that would be a useful addition. Can we raise it as a separate task though please since it is out of scope for this issue.

Reply:
Yes, this is not a must here.